src/shogun/util/factory.h

```diff
@@ -130,6 +131,41 @@ namespace shogun
 		return result;
 	}
 
+	template <class T>
+	std::shared_ptr<Features> sparse_features(SGMatrix<T> mat)
```

Comment:
exposing the SparseMatrix<T> ctor would be good as well, but as for factory we would like to have this under one common function, namely `features`.